File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        s = sorted(list(set(nums)))
        start = 0
        end = 0
        longest = 0
        while start < len(s):
            end = start
            while end + 1 < len(s) and s[end + 1] - s[end] == 1:
                end += 1
            longest = max(longest, end - start + 1)
            start = end + 1
        return longest

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> counts;
        for(int i = 0; i < nums.size(); i++){
            counts[nums[i]] += 1;
            if(counts[nums[i]] > 1){
                return true;
            }
        }
        return false;
        
    }
};